packages/web/src/contexts/AuthContext.js

```diff
@@ -77,24 +127,76 @@ const AuthContext = createContext({
 	...INITIAL_STATE,
 	login: () => {},
 	logout: () => {},
+	register: () => {},
+	validateEmail: () => {},
+	validateUsername: () => {},
 })
 
 const AuthContextProvider = props => {
 	const { children } = props
 	const [state, dispatch] = useReducer(reducer, { ...INITIAL_STATE })
 
+	const validateEmail = useCallback(async email => {
+		let response = null
+
+		try {
+			response = await fetch(`/api/users/validate-email?email=${email}`)
+		} catch (error) {}
+
+		if (response.status === 204) {
+			return true
+		}
+
+		return false
+	}, [])
+
+	const validateUsername = useCallback(async username => {
+		let response = null
+
+		try {
+			response = await fetch(`/api/users/validate-username?username=${username}`)
+		} catch (error) {}
+
+		if (response.status === 204) {
+			return true
+		}
+
+		return false
+	}, [])
+
+	const handleAuthStateChanged = useCallback(user => {
+		dispatch({
+			payload: user,
+			type: 'auth state changed',
+		})
+	}, [dispatch])
+
+	const handleProfileSnapshot = useCallback(snapshot => {
+		dispatch({
+			payload: snapshot.data(),
+			type: 'profile loaded',
+		})
+	}, [dispatch])
+
+	const handleSettingsSnapshot = useCallback(snapshot => {
+		dispatch({
+			payload: snapshot.data(),
+			type: 'settings loaded',
+		})
+	}, [dispatch])
+
 	const login = useCallback(async (email, password) => {
 		dispatch({ type: 'attempt login' })
 
 		try {
-			// await auth.login()
+			await auth.signInWithEmailAndPassword(email, password)
 			return dispatch({ type: 'login success' })
 		} catch (error) {
 			return dispatch({ type: 'login failure' })
 		}
 	}, [dispatch])
 
-	const logout = useCallback(async (email, password) => {
+	const logout = useCallback(async () => {
 		dispatch({ type: 'attempt logout' })
 
 		try {
```

Comment:
await auth.logout() is commented out so it seems we would always return true here, Is this desired?

Reply:
For the moment, yes. I haven't actually implemented logout yet. For dev, we can logout by clearing our local application state via the devtools. Logout will be added with my next PR, though. 😁